osu.Game/Screens/OnlinePlay/Lounge/Components/RoomsContainer.cs

```diff
@@ -137,7 +137,9 @@ private void removeRooms(IEnumerable<Room> rooms)
 
                 roomFlow.Remove(toRemove);
 
-                selectedRoom.Value = null;
+                // selection may have a lease due to being in a sub screen.
+                if (!selectedRoom.Disabled)
+                    selectedRoom.Value = null;
```

Comment:
There are more writes to `selectedRoom` in this class than this one, and they can crash because they're bound to UI items. `OnClick()` may be mostly safe thanks to the loading layer in front, but `selectNext()` can most definitely blow up.

This would be best reproducible by attaching a delay between `LoungeSubScreen` taking the lease and the room subscreen being pushed, and then spamming arrow up/down during that delay. I used the debugger to help me with that, but it sure does crash.

Reply:
Maybe we can remove the global selected room bindable... After the lounge redesign, it'll only be used for the multiplayer background but perhaps it's possible to simplify that flow without using a bindable.

Reply:
Do you think it's worth doing that refactor here? At very least, it would require moving the scroll container inside `RoomsContainer` as it is currently being used there.

Reply:
It may be slightly too complex to refactor as part of this PR.

Reply:
To avoid growing this PR any further (and just fix the underlying issue, which is a pretty serious one) I've stuck to fixing the remaining two operations in a hope to get it in a release today.